src/main/kotlin/lotto/domain/LottoResult.kt

```diff
@@ -1,46 +1,30 @@
 package lotto.domain
 
 class LottoResult(
-    private val winningNums: List<LottoNumber>,
+    private val winningLotto: WinningLotto,
     private val lottos: List<Lotto>
 ) {
-    private val resultMap = mutableMapOf<LottoRank?, Int>()
-    var rateOfReturn: Double = 0.0
+    val rankCntMap = mutableMapOf<LottoRank, Int>()
+    var rateOfReturn = 0.0
 
     init {
         setResultMap()
-        setReturn(getTotalProfit())
-    }
-
-    override fun toString(): String {
-        var result = ""
-        LottoRank.ranks().forEach {
-            val cnt = resultMap.getOrDefault(it, 0)
-            println("${it.numOfMatch}" + "개 일치 (${it.winningMoney}" + "원)- $cnt" + "개\n")
-        }
-        result += "총 수익률은 ${String.format("%.2f", rateOfReturn)}" + "입니다."
-        result += if (rateOfReturn < 1) LOSS else GAIN
-        return result
+        setReturnRate(getTotalProfit())
     }
 
     private fun setResultMap() {
         lottos.forEach {
-            val numOfMatch = it.numOfMatch(winningNums)
-            val rank = LottoRank.getRankByNumOfMatch(numOfMatch)
-            resultMap[rank] = resultMap.getOrDefault(rank, 0) + 1
+            val rank = winningLotto.rankOfLotto(it)
+            rankCntMap[rank] = rankCntMap.getOrDefault(rank, 0) + 1
```

Comment:
elvis 연산자를 활용하면 다음과 같이 표현할 수도 있습니다!
```suggestion
            rankCntMap[rank] = (rankCntMap[rank] ?: 0) + 1
```